s3simple

```diff
@@ -71,7 +72,14 @@ s3simple() {
   local bucket="${path%/*}"
   local key="${path#*/}"
 
-  curl "${args[@]}" -s -f -H "Date: ${date}" -H "Authorization: ${authorization}" "https://${bucket}.s3.amazonaws.com/${key}"
+  local url
+  if [ z "${AWS_REGION-}" ]; then
+    url="s3-${AWS_REGION}.amazonaws.com"
+  else
+    url="s3.amazonaws.com"
+  fi
+
+  curl "${args[@]}" -s -f -H "Date: ${date}" -H "Authorization: ${authorization}" "https://${bucket}.${url}/${key}"
```

Comment:
```suggestion
  local domain
  if [ z "${AWS_REGION-}" ]; then
    domain="s3-${AWS_REGION}.amazonaws.com"
  else
    domain="s3.amazonaws.com"
  fi

  curl "${args[@]}" -s -f -H "Date: ${date}" -H "Authorization: ${authorization}" "https://${bucket}.${domain}/${key}"
```

Reply:
I think `domain` would be a better name for this variable - the whole thing is a URL, this is a component of the hostname so domain seems to make the most sense to me